File: packages/iscp/receipt.go
package iscp

import "github.com/iotaledger/wasp/packages/vm/gas"

// Receipt represents a blocklog.RequestReceipt with translated error string
type Receipt struct {
	Request         []byte             `json:"request"`
	Error           *UnresolvedVMError `json:"error"`
	GasBudget       uint64             `json:"gasBudget"`
	GasBurned       uint64             `json:"gasBurned"`
	GasFeeCharged   uint64             `json:"gasFeeCharged"`
	BlockIndex      uint32             `json:"blockIndex"`
	RequestIndex    uint16             `json:"requestIndex"`
	TranslatedError string             `json:"translatedError"`
	GasBurnLog      *gas.BurnLog       `json:"-"`
}

File: packages/wasmvm/wasmclient/src/isc/receipt.rs
pub struct Receipt {
    request: &'static [u8],
    error: Result<(), &'static str>,
    gas_budget: u64,
    gas_burned: u64,
    gas_fee_charged: u64,
    block_index: u32,
    request_index: u16,
    resolved_error: &'static str,
    gas_burn_log: *const BurnLog,
}

type BurnCode = u16;

pub struct BurnRecord {
    code: BurnCode,
    gas_burned: u64,
}

pub struct BurnLog {
    records: [BurnRecord],
}
